src/main/java/core/mate/academy/model/ExcavatorProducer.java

```diff
@@ -0,0 +1,20 @@
+package core.mate.academy.model;
+
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer {
```

Comment:
we should parametrize MachineProducer

Reply:
Same question as above, could you explain, please. We need to add \<T extends Machine\> after ExcavatorProducer ? I did so, but T is never used inside the class and if we do use it, we get a mistake that expected type is Excavator, but T was received